doc/specs/#888 - Com Api.md

```diff
@@ -0,0 +1,889 @@
+        /// User means it was added by the user and could be removed.
+        User,
+    };
+
+    /// IMPLEMENTATION NOTE: SourceTrustLevel from AppInstallerRepositorySource.h
+    /// Defines the trust level of the app catalog.
+    [contract(Microsoft.Management.Deployment.WindowsPackageManagerContract, 1)]
+    enum AppCatalogTrustLevel
+    {
+        None,
+        Trusted,
+    };
+
+    /// IMPLEMENTATION NOTE: SourceDetails from AppInstallerRepositorySource.h
+    /// Interface for retrieving information about an app catalog without acting on it.
+    [contract(Microsoft.Management.Deployment.WindowsPackageManagerContract, 1)]
+    runtimeclass AppCatalogInfo
+    {
+        /// The app catalog's unique identifier.
+        String Id { get; };
+        /// The name of the app catalog.
+        String Name { get; };
+        /// The type of the app catalog.
+        String Type { get; };
+        /// The argument used when adding the app catalog.
+        String Arg { get; };
+        /// The app catalog's extra data string.
+        String ExtraData { get; };
+        /// The last time that this app catalog was updated.
+        Windows.Foundation.DateTime LastUpdateTime { get; };
+        /// The origin of the app catalog.
+        AppCatalogOrigin Origin { get; };
+        /// The trust level of the app catalog
+        AppCatalogTrustLevel TrustLevel { get; };
+    };
+
+    /// A metadata item of a package version.
+    [contract(Microsoft.Management.Deployment.WindowsPackageManagerContract, 1)]
+    enum PackageVersionMetadata
+    {
+        /// The InstallerType of an installed package
+        InstalledType,
+        /// The Scope of an installed package
+        InstalledScope,
+        /// The system path where the package is installed
+        InstalledLocation,
+        /// The standard uninstall command; which may be interactive
+        StandardUninstallCommand,
+        /// An uninstall command that should be non-interactive
+        SilentUninstallCommand,
+        /// The publisher of the package
+        Publisher,
+        /// The locale of the package
+        Locale,
+    };
+
+    /// IMPLEMENTATION NOTE: IPackageVersion from AppInstallerRepositorySearch.h
+    /// A single package version.
+    [contract(Microsoft.Management.Deployment.WindowsPackageManagerContract, 1)]
+    runtimeclass PackageVersionInfo
+    {
+        /// IMPLEMENTATION NOTE: PackageVersionMetadata fields from AppInstallerRepositorySearch.h
+        /// Gets any metadata associated with this package version.
+        /// Primarily stores data on installed packages.
+        /// Metadata fields may have no value (e.g. packages that aren't installed will not have an InstalledLocation).
+        String GetMetadata(PackageVersionMetadata metadataType);
+        /// IMPLEMENTATION NOTE: PackageVersionProperty fields from AppInstallerRepositorySearch.h
+        String Id { get; };
+        String Name { get; };
+        String Version { get; };

```

Comment:
"3.14" and "3.4" can be compared as strings but may not produce the expected result. SemVer has similar issues e.g. SemVer says sort order is
```
3.14-alpha
3.14-beta
3.14-delta1
3.14-delta2
3.14-delta10
```
but compared as simple strings will sort differently. Can relational comparisons be performed always, never or sometimes (depending on the package, i.e. buyer-beware)?

Similarly, can values be compared for equality? `3.14 == 3.14.0'? "October 2022 Refresh" == "october 2022 refresh"? Are there rules for equality testing e.g. lower-case any ASCII characters? Unicode introduces lots of fun potential equivalencies (ae vs æ, e' vs é, etc), depending on the rules you follow.

If comparison rules aren't simply ordinal are additional properties/methods needed? e.g. if version is numeric and "3.14" and "3.14.0" are both legal and equivalent, should there be a `int64 VersionAsDotQuadNumber { get; }` returning "3.14" and "3.14.0" as 0x0003000D00000000?

Reply:
I get the impression that the Version is "whatever the app wants it to be". There are conventions that most apps follow, but no hard-and-fast rules. If you know what version scheme a specific app follows, you can parse it according to that scheme. Otherwise, it's "parse at your own risk."